src/infinitempomatrix.jl

```diff
@@ -44,111 +44,212 @@ function InfiniteMPOMatrix(data::Vector{Matrix{ITensor}}, translator::Function)
   return InfiniteMPOMatrix(CelledVector(data, translator), 0, size(data)[1], false)
 end
 
-#
-#  Hm should have the form below.  Only link indices are shown.
-#
-#    I         0                     0           0      0
-#  M-->l=n     0                     0           0      0
-#    0    l=n-->M-->l=n-1 ...        0           0      0
-#    :         :                     :           :      :
-#    0         0          ...  l=2-->M-->l=1     0      0
-#    0         0          ...        0        l=1-->M   I
-#
-#  We need to capture the corner I's and the sub-diagonal Ms, and paste them together into an ITensor.
-#  This is facilutated by making all elements of Hm into order(4) tensors by adding dummy Dw=1 indices.
-#  We ITensors.directsum() to join all the blocks into the final ITensor.
-#  This code should be 100% dense/blocks-sparse agnostic.
-#
-function cat_to_itensor(Hm::Matrix{ITensor}, inds_array)::Tuple{ITensor,Index,Index}
+function InfiniteMPOMatrix(data::CelledVector{Matrix{ITensor}}, m::Int64, n::Int64)
+  return InfiniteMPOMatrix(data, m, n, false)
+end
+
+function InfiniteMPOMatrix(data::CelledVector{Matrix{ITensor}})
+  return InfiniteMPOMatrix(data, 0, size(data)[1], false)
+end
+
+function ITensors.siteinds(A::InfiniteMPOMatrix)
+  return CelledVector(
+    [dag(only(filterinds(A[x][1, 1]; plev=0, tags="Site"))) for x in 1:nsites(A)],
```

Comment:
We should avoid using `filterinds` with `tags`, not all site indices will have `"Site"` tags.